### PR TITLE
Integrate Navigation library

### LINKED ISSUE #23: Integrate Navigation library
### Description

We should integrate some way of handling navigation between screens. The first approach we might want to try is one recommended in the official documentation: https://reactnative.dev/docs/navigation

### PR DESCRIPTION
**Pull request by wzieba**
Closes #23 

See internal post for details: pe5pgL-394-p2

## Description

This PR integrates `react-router-native` package to help us with handling navigation.

### CODE REVIEW

File: ShippingZones.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+              caption={item.methods.map((method) => method.title).join(" - ")}
+            />
+          )}
+          keyExtractor={(item) => item.id}
+        />
+      )}
+      <SafeAreaView>
+        <Button
+          title="+ Add new Shipping Zone"
+          onPress={() => navigate("/addShippingZone")}
```
Comment (Ecarrion) [inline ShippingZones.tsx]: Do you think is worth having these paths in some centralized place? It seems easy to mistype them 😅

**Review by Ecarrion — APPROVED** [review]
Hey @wzieba! Thanks for integrating this library.

I added the Some `SafeAreaView` wrappers so the content is not hidden on iOS by the notch and by the home indicator.

Before | After
---- | ----
before-1 | after-1
before-2 | after-2

---- 

Additionally, I noticed that console was throwing a "can't store values" error. 

This was happening because we split the components for this new navigation, we weren't passing the properties injected on the main component to the new `ShippingZonesList` subcomponent.

Instead of fixing this by passing the properties, I decided that is safer to store the dependencies on the main component rather than in a subcomponent.

The only complication here was that we need to ensure that the component that reads data (ShippingZonesList) is not rendered until the injected properties are stored. To do this we have to add a `saving` state in the main component.

**Comment by wzieba** [comment]
hi @Ecarrion 👋 Thank you for adding improvements and fixing the bug, great work!

As discussed on P2, I've decided to move back to React Navigation. So far, I like it more that react-router for these reasons:
- as you said, it has navigation animations out of the box. Although it looks bad on Android (too slow). But I'd consider this as a minor for now.
- out of the box native navigation toolbar
- it doesn't recreate screens on navigating back. See that list of Shipping Zones is not re-fetched.

Could you check how things work on iOS?


https://github.com/woocommerce/WooCommerce-Shared/assets/5845095/4366d3eb-d875-4d9c-ae83-0a2c55129c38

**Review by Ecarrion — APPROVED** [review]
This worked well for me!

Tested iOS and Android! ✅

**Pull request merged**